### PR TITLE
fix(api): 🐛 Make case expiry 45 days (1080 hours) instead of 72 hours

### PR DESCRIPTION
**Pull request by Sven65**
Depends on https://github.com/helsingborg-stad/helsingborg-io-sls-resources/pull/52

## Feature description

Makes the case expiry constant 1080 hours (45 days * 24 hours) instead of 72 hours as described in #CU-hd9zcf
I've also added a new constant in order to keep the 72 hours for newly created cases.

## Solution description

- Added constant `CASE_ONGOING_EXPIRATION_HOURS`
- Rename `CASE_EXPIRATION_HOURS` to `CASE_SUBMITTED_EXPIRATION_HOURS`
- Add switch case in `updateCase` to check the status of the case update in order to figure out how (and if) to update the TTL

## What areas is affected by these changes?.

Cases expiry

## Is there any existing behavior change of other features due to this code change?

No

## Are your code strutured in a way so that reviewers can understand it?

Yes

## Was this feature tested in the following environments?
- [x] Your personal AWS environment.
- [] Your local Serverless environment.

### CODE REVIEW

**Comment by helsingborgsstad** [comment]
Task linked: [CU-hd9zcf När ärende har status inskickat ska expiry date (TTL) ha 45 dagar](https://app.clickup.com/t/hd9zcf)

File: services/cases-api/lambdas/updateCase.js
Comment (dannil76) [inline services/cases-api/lambdas/updateCase.js:70]: Hi Max,
At this moment the expire time is only relevant for cases with the provider 'VIVA'

My suggestion is that you extract this logic in a separate lambda and put it under the viva-ms service.

Then make the lambda trigger on status: active:ongoing or active:submitted:viva. See serverless.yml (viva-ms) for more help in setting the event bridge params

I'd be glad if we could replace the switch statement for a function call as well :D

Just poke me if this make no sense.

**Review by dannil76 — CHANGES_REQUESTED** [review]

File: services/cases-api/lambdas/updateCase.js
Comment (Sven65) [reply services/cases-api/lambdas/updateCase.js]: Moved it to its own lambda, and removed the switch statement.

**Review by dannil76 — COMMENTED** [review]
Looks very good. Have one wish. Move lambda main function to the top of file to flow how other lambdas is implemented. :)

Then I struck me. I think we have to rethink the event logic a bit. Let me come back.

File: services/viva-ms/lambdas/syncExpiryTime.js
Comment (TeddyAndersson) [inline services/viva-ms/lambdas/syncExpiryTime.js:75]: Since this is a very general function that we probably use in more places than this. I think this belongs in the libs folder under dynamoDb or queries.

File: services/viva-ms/lambdas/syncExpiryTime.js
Comment (TeddyAndersson) [inline services/viva-ms/lambdas/syncExpiryTime.js:66]: The definition of hours here is a bit confusing, since it's easy to miss the last line where you retrive the hours value. My suggestion is to simplify the code by defining the object and then retriving the value. The object could then also be set outside of the function since it won't change.

```
const statusToHours = {
    'active:ongoing': VIVA_CASE_ONGOING_EXPIRATION_HOURS,
    'active:submitted:viva': VIVA_CASE_SUBMITTED_EXPIRATION_HOURS,
}

const hours = statusToHours[status]
```

**Review by TeddyAndersson — CHANGES_REQUESTED** [review]
Well done, i think the solution in general is solid. Added some minor comments regarding code readability. 👍  😄

**Comment by dannil76** [comment]
Realize that this will not work. Lambda is triggered as soon as the case is updated in the database. Yep all peace and joy so far.

But! When lambda updates the case with a new expiration date, the lambda is triggered again and again when the database is updated and the lambda listens all the time.

Have a solution in progress.

File: services/cases-api/lambdas/createCase.js
Comment (dannil76) [reply services/cases-api/lambdas/createCase.js]: Fixed

File: libs/constants.js
Comment (dannil76) [reply libs/constants.js:7]: Fixed

File: services/viva-ms/lambdas/createVivaCase.js
Comment (dannil76) [reply services/viva-ms/lambdas/createVivaCase.js]: Fixed

File: services/viva-ms/lambdas/syncExpiryTime.js
Comment (dannil76) [reply services/viva-ms/lambdas/syncExpiryTime.js]: Removed

File: services/viva-ms/lambdas/syncExpiryTime.js
Comment (dannil76) [reply services/viva-ms/lambdas/syncExpiryTime.js]: Not follow

File: services/viva-ms/lambdas/syncExpiryTime.js
Comment (dannil76) [reply services/viva-ms/lambdas/syncExpiryTime.js]: Jepp. Some other day

File: services/viva-ms/lambdas/syncExpiryTime.js
Comment (dannil76) [reply services/viva-ms/lambdas/syncExpiryTime.js]: Fixed

File: services/viva-ms/lambdas/syncExpiryTime.js
Comment (TeddyAndersson) [inline services/viva-ms/lambdas/syncExpiryTime.js:30]: Suggestion for change, this if statement could be used when getting an item from the DynamoDB response.

const caseItem = getCaseResponse.Items.find(item => item.currentFormId === vivaCaseSSMParams.recurringFormId)

if (!caseItem) {
   console.info('(Viva-ms: syncExpiryTime): case is not an recurring form. Nothing to update.');
   return true;
}

File: services/viva-ms/lambdas/syncExpiryTime.js
Comment (TeddyAndersson) [inline services/viva-ms/lambdas/syncExpiryTime.js:28]: '(Viva-ms: syncExpiryTime): case is not an recurring form. Nothing to update.' -> '(Viva-ms: syncExpiryTime): case attribute currentFormId (caseItem.currentFormId) does not match the recurring form id (vivaCaseSSMParams.recurringFormId) from ssm params. Nothing to update.'

A bit more detailed information in my opinion :)

**Review by TeddyAndersson — COMMENTED** [review]
Left some comments for improvements, but in general the code looks fine to me. Will need to test this aswell before approval

File: services/viva-ms/lambdas/syncExpiryTime.js
Comment (dannil76) [reply services/viva-ms/lambdas/syncExpiryTime.js:24]: Hmm. not sure if I agree with you about readbility. Items is a list and I grab the first thing from it.
Why complicate things with a more complex expression?

File: services/viva-ms/lambdas/syncExpiryTime.js
Comment (dannil76) [reply services/viva-ms/lambdas/syncExpiryTime.js:30]: hmm nja

File: services/viva-ms/lambdas/syncExpiryTime.js
Comment (dannil76) [reply services/viva-ms/lambdas/syncExpiryTime.js:28]: to verbose in my opinion

File: services/viva-ms/lambdas/syncExpiryTime.js
Comment (TeddyAndersson) [reply services/viva-ms/lambdas/syncExpiryTime.js:24]: We should trust that getCase is doing it's thing. or?
This is the question we should ask ourself i guess in regards for the Lambda, so should the lambda trust that DynamoDb does what it's suppose to do? I would argue that it's better to not trust DynamoDb, thereby a check to se if the case that is returned is the one we wanted. 


Nevermind the readability part mentioned in my initial comment.

File: services/viva-ms/lambdas/syncExpiryTime.js
Comment (TeddyAndersson) [reply services/viva-ms/lambdas/syncExpiryTime.js:28]: Then we need a second opinon. My take is that rather log a detailed description telling the reader what broke so that we know in detail why the lambda fail. In other words I rather have a more verbose logging to help me while debugging an issue :)

**Review by TeddyAndersson — APPROVED** [review]
Looks good and behaves as intended, from the infromation given.

**Pull request merged**
